api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -0,0 +1,82 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { intersection } from 'lodash';
+
+import { UsersProjectsApiEntity } from '@marxan-api/modules/projects/control-level/users-projects.api.entity';
+import { Roles } from '@marxan-api/modules/users/role.api.entity';
+
+import { Permit, ProjectAccessControl } from './project-access-control';
+
+/**
+ * Debt: neither UsersProjectsApiEntity should belong to projects
+ * nor the Roles should belong to users
+ */
+@Injectable()
+export class ProjectAclService implements ProjectAccessControl {
+  private readonly canCreateScenarioRoles = [Roles.project_owner];
+  private readonly canEditScenarioRoles = [Roles.project_owner];
+  private readonly canViewSolutionRoles = [Roles.project_owner];
+  private readonly canPublishProjectRoles = [Roles.project_owner];
+  private readonly canViewProjectRoles = [
+    Roles.project_owner,
+    Roles.project_contributor,
+    Roles.project_viewer,
+  ];
+
+  private async getRolesWithinProject(
```

Comment:
you may have noticed already that I am fastidious about naming things 😱  - tiny detail, and feel free to ignore: what about the slightly more verbose name below?

```suggestion
  private async getRolesWithinProjectForUser(
```